web/packages/teleport/src/Discover/Shared/SetupAccess/useUserTraits.test.tsx

```diff
@@ -587,6 +587,46 @@ describe('static and dynamic traits are correctly separated and correctly create
   });
 });
 
+describe('calls to nextStep respects number of steps to skip', () => {
+  test('with auto discover, as a sso user with no traits', async () => {
+    const teleCtx = createTeleportContext();
+    teleCtx.storeUser.state.authType = 'sso';
+    const user = getMockUser();
+    user.traits = {
+      logins: ['login'],
+      databaseUsers: [],
+      databaseNames: [],
+      kubeUsers: [],
+      kubeGroups: [],
+      windowsLogins: [],
+      awsRoleArns: [],
+    };
+
+    jest.spyOn(teleCtx.userService, 'fetchUser').mockResolvedValue(user);
+
+    const discoverCtx = defaultDiscoverContext({
+      resourceSpec: defaultResourceSpec(ResourceKind.Database),
+    });
+    discoverCtx.agentMeta = {
+      ...discoverCtx.agentMeta,
+      autoDiscovery: { config: { name: '', discoveryGroup: '', aws: [] } },
+    };
```

Comment:
Why not just `discoverCtx.agentMeta.autoDiscovery = {...}`?

Reply:
copy paste gone wrong